components/reddit/sources/new-comments-by-user/new-comments-by-user.js

```diff
@@ -0,0 +1,95 @@
+const common = require("../../common");
+const { reddit } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-comments-by-user",
+  name: "New comments by user",
+  description: "Emits an event each time a user posts a new comment.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    username: { propDefinition: [reddit, "username"] },
+    context: {
+      type: "integer",
+      label: "context",
+      description: "an integer between 2 and 10",
+    },
+    t: { propDefinition: [reddit, "t"] },
+    sr_detail: { propDefinition: [reddit, "sr_detail"] },
+    ...common.props,
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      let before = null;
+      try {
+        var reddit_comments = await this.reddit.getNewUserComments(
+          null,
+          this.username,
+          this.context,
+          this.t,
+          this.sr_detail,
+          10
+        );
+      } catch (err) {
+        if (common.methods.did4xxErrorOccur(err)) {
+          throw new Error(
+            `We encountered a 4xx error trying to fetch comments by ${this.username}. Please check the username and try again`
+          );
+        }
+        throw err;
+      }
+      var reddit_comments_pulled = common.methods.wereThingsPulled(
+        reddit_comments
+      );
+      if (reddit_comments_pulled) {
+        before = reddit_comments.data.children[0].data.name;
+        const ordered_reddit_comments = reddit_comments.data.children.reverse();
+        ordered_reddit_comments.forEach((reddit_comment) => {
+          this.emitRedditEvent(reddit_comment);
+        });
+      }
+      this.db.set("before", before);
```

Comment:
I think we can clean this up a bit, and get rid of the `wereThingsPulled` method:
```js
const { children: comments = [] } = reddit_comments.data;
if (comments.length === 0) {
  console.log("No new comments found. Skipping...");
  return;
}

const {
  name: before = this.db.get("before"),
} = comments[0].data;
this.db.set("before", before);

comments
  .reverse()
  .forEach(this.emitRedditEvent.bind(this));
```